backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/DockerCassandra.java

```diff
@@ -28,12 +31,110 @@
 import org.testcontainers.images.builder.dockerfile.DockerfileBuilder;
 
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.schemabuilder.SchemaBuilder;
 import com.github.dockerjava.api.DockerClient;
 import com.google.common.collect.ImmutableMap;
 
 public class DockerCassandra {
 
+    /**
+     * James uses a non privileged Cassandra user(role) in testing. To be able to do that, the non privileged user needs to be
+     * prepared along with a created keyspace.
+     *
+     * This process is done by using the default user provided by docker cassandra, it has the capability of creating roles,
+     * keyspaces, and granting permissions to those entities.
+     */
+    public static class CassandraResourcesManager {
+
+        private static final String CASSANDRA_SUPER_USER = "cassandra";
+        private static final String CASSANDRA_SUPER_USER_PASSWORD = "cassandra";
+
+        private final DockerCassandra cassandra;
+
+        private CassandraResourcesManager(DockerCassandra cassandra) {
+            this.cassandra = cassandra;
+        }
+
+        public void initializeKeyspace(String keyspace) {
+            ClusterConfiguration configuration = configurationBuilder(keyspace).build();
+
+            try (Cluster privilegedCluster = ClusterFactory.create(configuration)) {
+                provisionNonPrivilegedUser(privilegedCluster);
+                KeyspaceFactory.createKeyspace(configuration, privilegedCluster);
+                grantPermissionTestingUser(privilegedCluster, keyspace);
+            }
+        }
+
+        public void dropKeyspace(String keyspace) {
+            try (Cluster cluster = ClusterFactory.create(configurationBuilder(keyspace).build())) {
+                try (Session cassandraSession = cluster.newSession()) {
+                    boolean applied = cassandraSession.execute(
+                        SchemaBuilder.dropKeyspace(keyspace)
+                            .ifExists())
+                        .wasApplied();
+
+                    if (!applied) {
+                        throw new IllegalStateException("cannot drop keyspace '" + keyspace + "'");
+                    }
+                }
+            }
+
+        }
+
+        public boolean keyspaceExist(String keyspace) {
+            try (Cluster cluster = ClusterFactory.create(configurationBuilder(keyspace).build())) {
+                return KeyspaceFactory.keyspaceExist(cluster, keyspace);
+            }
+        }
+
+        private void provisionNonPrivilegedUser(Cluster privilegedCluster) {
+            try (Session session = privilegedCluster.newSession()) {
+                session.execute("CREATE ROLE IF NOT EXISTS " + CASSANDRA_TESTING_USER + " WITH PASSWORD = '" + CASSANDRA_TESTING_PASSWORD + "' AND LOGIN = true");
+            }
+        }
+
+        private void grantPermissionTestingUser(Cluster privilegedCluster, String keyspace) {
```

Comment:
`grantPermissionToTestingUser`

Reply:
(Maybe having a strong type for keyspace could be cool?)

Reply:
In another PR? That's not simple, the keyspace contains name, durableWrites and a map of repication options.